internal/apiserver/createInvoice.go

```diff
@@ -88,69 +35,80 @@ func validateInvoice(ctx goContext.Context, db *db.Connector, inv *entity.Invoic
 	}
 }
 
-func (a *App) createInvoice(res http.ResponseWriter, req *http.Request) error {
-	requestBody := CreateInvoiceRequestBody{}
-	err := requestBody.parse(req)
-	if err != nil {
-		return err
-	}
-	userId := context.GetUserId(req.Context())
+func (a *App) createInvoice(testInvoice bool) func(res http.ResponseWriter, req *http.Request) error {
+	return func(res http.ResponseWriter, req *http.Request) error {
+		invoice, err := parseInvoice(req)
```

Comment:
let's extract this top part to function so we can reuse it for visualization as well